### ISSUE TITLE
Support for exposing fragment parameters in amp-analytics

### ISSUE THREAD

**Issue by rudygalfi** [ampproject/amphtml#2139]
Offshoot from #1299, where we added support for query params but didn't cover fragment params.

This would cover the case where in a page URL like foo.com/section/page.html#b=3, you could use a variable substitution like FRAGMENT_PARAM(b) and it would get replaced by "3"

**Comment by adelinamart** [comment]
@rudygalfi is this something we are still looking into? Thanks

**Comment by rudygalfi** [comment]
Will close this for now and wait for a specific use case to come along. Note that #3135 does this for a very specific use case (share tracking).

**Comment by johnboxall** [comment]
I believed I had a use-case for this, but in working on it, I realized we didn't.

_I'll leave what I found here in hopes it helps others._

> TLDR, you can probably use [`QUERY_PARAM`](https://github.com/ampproject/amphtml/blob/dcd136dcef84a168c774fa81de41bae79ae37dc9/spec/amp-var-substitutions.md#query-parameter) substitutions to achieve what you want.

We operate a server that responds with AMP pages. Let's call it `amp.example.com`.

Traffic arrives from ads with unique URLs, passed via anchor:

* `amp.example.com/#clickid=1`
* `amp.example.com/#clickid=2`
* `amp.example.com/#clickid=3`

We wanted to report `clickid` using `<amp-analytics>` but we found out anchors weren't supported in variable substitutions as per this issue.

To work around this we:

1) Changed ad URLs to pass `clickid` as a querystring parameter: `amp.example.com/?clickid=1`
2) Changed server's caching behaviour to ignore querystrings so as not to significantly impact our cache hit ratio (optional)

As substitutions run clientside, we could now access the `clickid` variable using `QUERY_PARAM(clickid)` and report analytics correctly!

_Hope this helps anyone else who is stuck!_

❤️
